src/include/btmem.h

```diff
@@ -757,12 +763,13 @@ struct __wt_ref {
 	volatile uint32_t pindex_hint;	/* Reference page index hint */
 
 #define	WT_REF_DISK	 0		/* Page is on disk */
-#define	WT_REF_DELETED	 1		/* Page is on disk, but deleted */
-#define	WT_REF_LOCKED	 2		/* Page locked for exclusive access */
-#define	WT_REF_LOOKASIDE 3		/* Page is on disk with lookaside */
-#define	WT_REF_MEM	 4		/* Page is in cache and valid */
-#define	WT_REF_READING	 5		/* Page being read */
-#define	WT_REF_SPLIT	 6		/* Parent page split (WT_REF dead) */
+#define	WT_REF_AMNESIA	 1		/* Page is in cache without history */
```

Comment:
My vote is for calling this new state `WT_REF_MEM_PARTIAL` - or `WT_REF_MEM_INCOMPLETE` if you have a taste for three extra characters.

Reply:
What I'm trying to capture with `WT_REF_AMNESIA` is the idea that we have a snapshot of the page without history.  We need to keep checking that the snapshot is valid for each reader of the page: if some reader needs the history, it has to instantiate it (and flip `ref->state` to `WT_REF_MEM`).

I don't love `MEM_PARTIAL` or `MEM_INCOMPLETE` because they suggest something is missing but don't have any connection to missing history.  They're almost negative flags ("REF_MEM" without something), which has gotten us confused in the past.

I spent ~too much~some time looking at a thesaurus: how about `WT_REF_LATENT` or `WT_REF_REPRESSED` or `WT_REF_BLANK_SLATE`?

Maybe less loaded, how about `WT_REF_IMAGE`, meaning "the disk image has been loaded without any history"?

Reply:
I think `WT_REF_IMAGE` is too generic. Although I don't like `WT_REF_AMNESIA` it is unique enough to catch one's attention.  I too spent a bit of time with the thesaurus, how about `WT_REF_LIMBO`? Defined at dictionary.com as `an intermediate, transitional, or midway state` which is a pretty accurate description of this, although it can have some connotations of being "incomplete". I see how you got to `AMNESIA` too.